tensorboard/uploader/uploader.py

```diff
@@ -140,6 +154,53 @@ def _upload_once(self):
         self._request_sender.send_requests(run_to_events)
 
 
+def update_experiment_metadata(
+    writer_client, experiment_id, name=None, description=None
+):
+    """Modifies user data associated with an experiment.
+
+    Args:
+      writer_client: a TensorBoardWriterService stub instance
+      experiment_id: string ID of the experiment to modify
+      name: If provided, modifies name of experiment to this value.
+      description: If provided, modifies the description of the experiment to
+         this value
+
+    Raises:
+      ExperimentNotFoundError: If no such experiment exists.
+      PermissionDeniedError: If the user is not authorized to modify this
+        experiment.
+      InvalidArgumentError: If the server rejected the name or description, if,
+        for instance, the size limits have changed on the server.
+    """
+    logger.info("Modifying experiment %r", experiment_id)
+    request = write_service_pb2.UpdateExperimentRequest()
+    request.experiment.experiment_id = experiment_id
+    if name is not None:
+        logger.info("Setting exp %r name to %r", experiment_id, name)
+        request.experiment.name = name
+        request.experiment_mask.name = True
```

Comment:
Not exactly in this PR, but it’s confusing that this field is not called
`update_mask` (per AIP 134); if it’s possible to change this before
stable launch, that would be great.

Reply:
Not doing this in this PR.  Let's discuss elsewhere if we can fit this in before stable launch.